doc/bootloader/README.md

```diff
@@ -0,0 +1,113 @@
+- A *BLE Stack* is a software library that implements the software functionalities needed to support BLE.
+- Finally, the *application firmware* is the actual firmware like InfiniTime.
+
+## The design
+
+Our goals when we designed the bootloader and the OTA procedure for the PineTime were:
+
+- to support multiple application firmwares;
+- to give as much freedom to firmware developers as possible;
+- to be as safe and reliable as possible.
+
+When designing the OTA procedure for the PineTime, we basically had 2 options : implement the OTA procedure in the bootloader or in the application firmware.
+
+Using the bootloader to handle the OTA is tempting as this would allow to implement it only once, and the application firmware would not need to care about it. However, the downside is that the bootloader would need to integrate the whole BLE stack.
+
+Indeed, the BLE stack is quite heavy in flash memory and a lot of memory would be waster if both the bootloader **and** the application firmware would integrate their own BLE stack. Of course, both the bootloader and the application firmware could share the same stack, but that's against our "freedom to the developers" goal : firmware developer would be forced to use whatever BLE stack we used in the bootloader.
+
+You've probably guessed it, we chose the 2nd solution, and we decided to implement the OTA procedure in the application firmware. In this case, the bootloader is much simpler, and firmware developers have a lot more freedom in their design choices (like the BLE stack to integrate, the data transfer protocol to implement,...).
+
+Here is basically how it works:
+
+- the application firmware exposes a BLE "firmware update" service. In InfiniTime we decided to implement the **nRF DFU** protocol.
+- this update service stores the new firmware in a temporary location in the *external flash memory* and checks the integrity of the data (CRC).
+- when the new firmware is completely transferred and checked, the watch is rebooted and the bootloader is ran again. The bootloader detects that a new firmware is available in the *external* flash memory and safely swaps the current firmware in the *internal* memory with the new one from the *external* memory.
+- When the swapping is done and checked, the new firmware is executed from the *internal* memory.
+- If the new firmware does not properly run, the bootloader will automatically revert to the older firmware by swapping again firmwares from the external to the internal memory.
+
+This procedure involves multiple software modules and rely on some hardware features. Let's go into more details!
+
+### Memory map
+
+Multiple memories are available in the PineTime : the internal RAM (64KB) and flash (512KB) memories from the nRF52832 and an external flash memory (4MB) connected to the SPI bus.
```

Comment:
```suggestion
Multiple memories are available in the PineTime:
the internal RAM (64KB) and flash (512KB) memories from the nRF52832 and an external flash memory (4MB) connected to the SPI bus.
```

Reply:
Perhaps to make the structure of the list easier to digest:
```suggestion
Multiple memories are available in the PineTime :
- 64KB of internal RAM on the nRF52832 microcontroller
- 512KB of internal flash memory on the nRF52832
- 4MB of external flash memory, which is connected to the SPI bus.
```